ui-core/src/components/inputs/Slider.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useState } from 'react';
+import type { ChangeEvent, MouseEvent, InputHTMLAttributes } from 'react';
+
+import cx from 'classnames';
+
+export type SliderProps = InputHTMLAttributes<HTMLInputElement> & {
+  width?: number;
+  onChangeCommitted?: (e: MouseEvent<HTMLInputElement>) => void;
+};
+
+// onChange returns an event or number
+const Slider = ({
+  id,
+  value: initialValue,
+  min = 0,
+  max = 100,
+  step = 1,
+  width,
+  onChange,
+  onChangeCommitted,
+  disabled,
+  ...rest
+}: SliderProps) => {
+  const [value, setValue] = useState<number>(
+    initialValue !== undefined ? Number(initialValue) : Number(min)
+  );
+
+  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const newValue = Number(e.target.value);
+    setValue(newValue);
+    onChange?.(e);
+  };
+
+  const handleMouseUp = (e: MouseEvent<HTMLInputElement>) => {
+    if (onChangeCommitted) {
+      onChangeCommitted(e);
+    }
+  };
+
+  return (
+    <div
+      className={cx('range-wrapper', { disabled })}
+      style={{ ...(width && { width: `${width}px` }) }}
```

Comment:
Maybe it's cleaner that way

```suggestion
  width = 112,
  onChange,
  onChangeCommitted,
  disabled,
  ...rest
}: SliderProps) => {
  const [value, setValue] = useState<number>(
    initialValue !== undefined ? Number(initialValue) : Number(min)
  );

  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
    const newValue = Number(e.target.value);
    setValue(newValue);
    onChange?.(e);
  };

  const handleMouseUp = (e: MouseEvent<HTMLInputElement>) => {
    if (onChangeCommitted) {
      onChangeCommitted(e);
    }
  };

  return (
    <div
      className={cx('range-wrapper', { disabled })}
      style={{ width: `${width}px` }}
```

Reply:
it’s nice, changed it